src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -498,7 +498,9 @@ public function dumpFile($filename, $content, $mode = 0666)
 
         if (!is_dir($dir)) {
             $this->mkdir($dir);
-        } elseif (!is_writable($dir)) {
+        }
+
+        if (!is_writable($dir)) {
```

Comment:
The handling was already there before. It just didn't take into account that the directory could have been created by another process.